src/tools/kata-ctl/src/check.rs

```diff
@@ -222,6 +232,71 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &str, parameter: &str) -> Result<String, String> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+    const MODULES_PATH: &str = "/sys/module";
+
+    // Partial check w/ modinfo
+    // verifies that the module exists
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module)
+        .output()
+        .expect("failed to execute process");
```

Comment:
@dborquez Was just looking at this code again, and I realized that this will again panic in case modinfo fails. Its better to return an error here with something like `map_err` and avoiding the `expect` call. 
cc @jodh-intel 

Similarly for the expect usage below.

Reply:
ok @amshinde let me check it out